lib/mpl_toolkits/axisartist/axis_artist.py

```diff
@@ -1007,13 +1032,19 @@ def _update_label(self, renderer):
             return
 
         if self._ticklabel_add_angle != self._axislabel_add_angle:
-            if ((self.major_ticks.get_visible()
-                 and not self.major_ticks.get_tick_out())
-                or (self.minor_ticks.get_visible()
-                    and not self.major_ticks.get_tick_out())):
-                axislabel_pad = self.major_ticks._ticksize
-            else:
-                axislabel_pad = 0
+            axislabel_pad = 0
+            if self.major_ticks.get_visible():
+                axislabel_pad = max(
+                    axislabel_pad,
+                    {"in": 1, "inout": .5, "out": 0}[
+                        self.major_ticks.get_tick_direction()]
+                    * self.major_ticks._ticksize)
+            if self.minor_ticks.get_visible():
+                axislabel_pad = max(
+                    axislabel_pad,
+                    {"in": 1, "inout": .5, "out": 0}[
+                        self.minor_ticks.get_tick_direction()]
+                    * self.major_ticks._ticksize)
```

Comment:
Overall maybe better?
```suggestion
            major_pad = (
                0 if not self.major_ticks.get_visible() else
                {"in": 1, "inout": .5, "out": 0}[self.major_ticks.get_tick_direction()]
                    * self.major_ticks._ticksize
            )
            minor_pad = (
                0 if not self.minor_ticks.get_visible() else
                {"in": 1, "inout": .5, "out": 0}[self.minor_ticks.get_tick_direction()]
                    * self.minor_ticks._ticksize)
            axislabel_pad = max(0, major_pad, minor_pad)
```